### ISSUE TITLE
Git with override `withLibsecret = true` fails to build, checks fail

### ISSUE THREAD

**Issue by spikespaz** [NixOS/nixpkgs#195891]
### Steps To Reproduce
1. build `pkgs.git.override {withLibsecret = true;}`

### Build log

**[Full Logs](https://termbin.com/zw8s)**

```
error: builder for '/nix/store/01516izzw990xv1ggqhbffljnjrfqy0i-git-2.37.3.drv' failed with exit code 2;
       last 20 log lines:
       > t9901-git-web--browse.sh ........................... ok
       > t9903-bash-prompt.sh ............................... ok
       > t9902-completion.sh ................................ ok
       >
       > Test Summary Report
       > -------------------
       > t0021-conversion.sh                              (Wstat: 256 (exited 1) Tests: 42 Failed: 1)
       >   Failed test:  31
       >   Non-zero exit status: 1
       > t0050-filesystem.sh                              (Wstat: 0 Tests: 11 Failed: 0)
       >   TODO passed:   9-10
       > t3910-mac-os-precompose.sh                       (Wstat: 256 (exited 1) Tests: 25 Failed: 10)
       >   Failed tests:  1, 4, 6, 8, 11-16
       >   TODO passed:   23
       >   Non-zero exit status: 1
       > Files=959, Tests=24405, 966 wallclock secs (15.30 usr  3.52 sys + 751.97 cusr 1737.84 csys = 2508.63 CPU)
       > Result: FAIL
       > make[1]: *** [Makefile:56: prove] Error 1
       > make[1]: Leaving directory '/build/git-2.37.3/t'
       > make: *** [Makefile:3065: test] Error 2
       For full logs, run 'nix log /nix/store/01516izzw990xv1ggqhbffljnjrfqy0i-git-2.37.3.drv'.
```

### Additional context
No additional context

### Notify maintainers
@primeos @wmertens @globin

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.19.14, NixOS, 22.11 (Raccoon), 22.11.20221009.34c5293`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by primeos** [comment]
Are you by any chance using ZFS? Apparently those two tests regularly cause issues when using ZFS or rather UTF-8 normalization: https://github.com/NixOS/nixpkgs/pull/191423

It should be fixed by https://github.com/NixOS/nixpkgs/pull/191423 either way though as that disables the two problematic tests. You can either cherry-pick that commit or wait until it lands in master / any channels: https://nixpk.gs/pr-tracker.html?pr=191423

Alternatively you can use `doInstallCheck = false` (I'd probably recommend that anyway as the tests take a significant amount of time, tend to fail for unrelated reasons, and I'm not sure if they've ever catched a Nixpkgs regression (they're more useful for upstream/development but we run them just in case)).

**Comment by spikespaz** [comment]
> Are you by any chance using ZFS? Apparently those two tests regularly cause issues when using ZFS or rather UTF-8 normalization: https://github.com/NixOS/nixpkgs/pull/191423
> 
> It should be fixed by https://github.com/NixOS/nixpkgs/pull/191423 either way though as that disables the two problematic tests. You can either cherry-pick that commit or wait until it lands in master / any channels: https://nixpk.gs/pr-tracker.html?pr=191423
> 
> Alternatively you can use `doInstallCheck = false` (I'd probably recommend that anyway as the tests take a significant amount of time, tend to fail for unrelated reasons, and I'm not sure if they've ever catched a Nixpkgs regression (they're more useful for upstream/development but we run them just in case)).

Yes, I use ZFS. I already disabled the install checks. This is good to know.

Not mecessarily going to concern myself with it, but can I cherry pick a commit with `overrideAttrs` using a new `src`?

Why is `doInstallCheck` true by default?
